foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/Oracle8Platform.java

```diff
@@ -14,12 +14,341 @@
+        return ClassConstants.CLOB.equals(type);
+    }
+
+    /**
+     * INTERNAL: Indicates whether app. server should unwrap connection to use
+     * lob locator.
+     */
+    public boolean isNativeConnectionRequiredForLobLocator() {
+        return true;
+    }
+
+    /**
+     * PUBLIC: Set if the locator is required for the LOB write. The default is
+     * true. For Oracle thin driver, the locator is recommended for large size (
+     * >4k for Oracle8, >5.9K for Oracle9) BLOB/CLOB value write.
+     */
+    public void setShouldUseLocatorForLOBWrite(boolean usesLocatorForLOBWrite) {
+        this.usesLocatorForLOBWrite = usesLocatorForLOBWrite;
+    }
+
+    /**
+     * PUBLIC: Return if the locator is required for the LOB write. The default
+     * is true. For Oracle thin driver, the locator is recommended for large
+     * size ( >4k for Oracle8, >5.9K for Oracle9) BLOB/CLOB value write.
+     */
+    public boolean shouldUseLocatorForLOBWrite() {
+        return usesLocatorForLOBWrite;
+    }
+
+    /**
+     * PUBLIC: Return the BLOB/CLOB value limits on thin driver. The default
+     * value is 0. If usesLocatorForLOBWrite is true, locator will be used in
+     * case the lob's size is larger than lobValueLimit.
+     */
+    public int getLobValueLimits() {
+        return lobValueLimits;
+    }
+
+    /**
+     * PUBLIC: Set the BLOB/CLOB value limits on thin driver. The default value
+     * is 0. If usesLocatorForLOBWrite is true, locator will be used in case the
+     * lob's size is larger than lobValueLimit.
+     */
+    public void setLobValueLimits(int lobValueLimits) {
+        this.lobValueLimits = lobValueLimits;
+    }
+
+    /**
+     * INTERNAL: Used by Oracle platforms during reading of ResultSet to free
+     * temporary LOBs.
+     */
+    public void freeTemporaryObject(Object value) throws SQLException {
+        if (value == null) {
+            return;
+        }
+
+        final Class valClass = value.getClass();
+        final String valClassName = valClass.getName();
+        if ("oracle.sql.CLOB".equals(valClassName)) {
+            final Class clobClass = resolveClass("oracle.sql.CLOB");
+            if (clobClass == null) {
+                return;
+            }
+            try {
+                final Method isTemporaryMethod = clobClass.getMethod("isTemporary");
+                isTemporaryMethod.invoke(value);
+            } catch (Exception e) {
+                // TODO: Emit Warning
+                return;
+            }
+
+        } else if ("oracle.sql.BLOB".equals(valClassName)) {
+            final Class blobClass = resolveClass("oracle.sql.BLOB");
+            if (blobClass == null) {
+                return;
+            }
+            try {
+                final Method isTemporaryMethod = blobClass.getMethod("isTemporary");
+                isTemporaryMethod.invoke(value);
+            } catch (Exception e) {
+                // TODO: Emit Warning
```

Comment:
Perhaps we can use something like this here: 
``` java
AbstractSessionLog.getLog().logThrowable(SessionLog.WARNING, null, e);
```

Reply:
Oh yeah, I left out the TODOs.  Good catch.